cpp/src/connection.cpp

```diff
@@ -93,6 +93,12 @@ session_range connection::sessions() const {
     return session_range(session_iterator(make_wrapper(pn_session_head(pn_object(), 0))));
 }
 
+std::string connection::url() const {
+    connection_context& cc = connection_context::get(pn_object());
+    if(!active()) throw proton::error("No active connection");
```

Comment:
I think this is an open question - what to do when the connection is not active: Could throw an exception as this code does, or somehow return nullptr maybe but then would need to change the return type and returning an pointer isn't a good idea overall. Or maybe return an option, but we don't do this anywhere else/

The issue here is whether it will be a bad user experience to receive an exception in the non active case.